app.go

```diff
@@ -618,12 +649,14 @@ func (app *App) startPubsubConnect() {
 					//reset backoff factor for successful connection
 					backoffFactor = 0
 
-					select {
-					case err = <-app.conn.Error:
-						app.close()
-						app.reportError(err)
-					case <-app.ctx.Done():
-						return
+					for _, connection := range app.conn {
+						select {
+						case err = <-connection.Error:
```

Comment:
> This will only block on one connection. We need to listen for all connections. May need something like this: https://stackoverflow.com/questions/19992334/how-to-listen-to-n-channels-dynamic-select-statement

Fixed